test/ngSanitize/sanitizeSpec.js

```diff
@@ -293,10 +293,56 @@ describe('HTML', function() {
     expect(doc).toEqual('<p><img src="x"></p>');
   }));
 
+  describe('Custom white-list support', function() {
+
+    var $sanitizeProvider;
+    beforeEach(module(function(_$sanitizeProvider_) {
+      $sanitizeProvider = _$sanitizeProvider_;
+
+      $sanitizeProvider.addValidElements(['foo']);
+      $sanitizeProvider.addValidElements({
+        htmlElements: ['foo-button', 'foo-video'],
+        htmlVoidElements: ['foo-input'],
+        svgElements: ['foo-svg']
+      });
+      $sanitizeProvider.addValidAttrs(['foo']);
+    }));
+
+    it('should allow custom white-listed element', function() {
+      expectHTML('<foo></foo>').toEqual('<foo></foo>');
+      expectHTML('<foo-button></foo-button>').toEqual('<foo-button></foo-button>');
+      expectHTML('<foo-video></foo-video>').toEqual('<foo-video></foo-video>');
+    });
+
+    it('should allow custom white-listed void element', function() {
+      expectHTML('<foo-input/>').toEqual('<foo-input>');
```

Comment:
What happens if the element passed in is not void? E.g. `<foo-input>some content</foo-input>`

Reply:
sanitizer removes closing tag

```
<foo-input>some content
```

Reply:
That is interesting...

Reply:
Yup, but this is how current sanitizer works.

When it meets token `<foo-inpupt>` it treats this as void element because it's valid case when self-closing element(void) does not have `/`, after that it meets text `some content` and we do nothing and finally we meet `</foo-input>` and because there are no open element before we just remove it.

Reply:
I will make it more explicit in the docs that void elements cannot have content.

Reply:
cannot or should not?

Reply:
Cannot and should not :grin:
As @maksimr mentioned `$sanitize` (and browsers) will transform `<void-element>foo</void-element>` to `<void-element>foo` (so the content ends up outside the element).